moto/transcribe/models.py

```diff
@@ -217,15 +221,27 @@ def advance(self) -> None:
                 "%Y-%m-%d %H:%M:%S"
             )
             if self._output_bucket_name:
-                transcript_file_uri = f"https://s3.{self._region_name}.amazonaws.com/{self._output_bucket_name}/"
-                if self.output_key is not None:
-                    transcript_file_uri += f"{self.output_key}/"
-                transcript_file_uri += f"{self.transcription_job_name}.json"
+                transcript_file_prefix = (
+                    f"https://s3.{self._region_name}.amazonaws.com/"
+                    f"{self._output_bucket_name}/"
+                    f"{(self.output_key or self.transcription_job_name).replace('.json', '')}"
```

Comment:
Bit of a nitpick - but `.json` could also be part of the filename (`this.is.not.a.json.file.xml`). So we should probably specifically check whether `.json` is at the end.

Reply:
You caught me in my laziness! Will update.

Reply:
K, updated the test to include a mid-key `.json` and only strip the suffix.